tests/experimental/datasources/test_spark_datasource.py

```diff
@@ -303,9 +303,9 @@ class TimeRange:
 
 
 def bad_regex_config(csv_path: pathlib.Path) -> tuple[re.Pattern, TestConnectionError]:
-    regex = re.compile(r"green_tripdata_sample_(?P<year>\d{4})-(?P<month>\d{2}).csv")
+    regex = re.compile(r"green_tripdata_sample_(?P<year>\d{4})-(?P<month>\d{2})\.csv")
     test_connection_error = TestConnectionError(
-        f"No file at path: {csv_path.resolve()} matched the regex: {regex.pattern}",
+        """No data references found in DataAsset "csv_spark_asset"."""
```

Comment:
We lost some of the useful information in this error message, can it be restored? I think it's helpful to know what the pattern was that failed the match